src/tests/BaGet.Core.Tests/FilePackageStorageServiceTests.cs

```diff
@@ -0,0 +1,17 @@
+using BaGet.Core.Services;
+using System.IO;
+using Xunit;
+
+namespace BaGet.Core.Tests
+{
+    public class FilePackageStorageServiceTests
+    {
+        [Fact]
+        public void CreatesStoreDirectoryIfItDoesNotExist()
+        {
+            string validButNotExistingPath = Path.Combine(Path.GetTempPath(), System.Guid.NewGuid().ToString("N"));
+            Assert.False(Directory.Exists(validButNotExistingPath));
+            IPackageStorageService service = new FilePackageStorageService(validButNotExistingPath);
+        }
```

Comment:
Should this have an `Assert.False(Directory.Exists(validButNotExistingPath));` after instantiation?

Reply:
currently this assertion throws a exception, because the constructor does NOT create the directoy (created later on demand)

should we create/ensure the directory during FilePackageStorageService construction ?

if we follow „fail fast and early“ probably yes....

if we do things like that inside the constructor:

-creating the directory verifies write permission.... nice
-should we do another check of write permission in case of directoryexists ?